src/libraries/Common/src/System/IO/PathInternal.CaseSensitivity.cs

```diff
@@ -12,47 +12,27 @@ namespace System.IO
     /// <summary>Contains internal path helpers that are shared between many projects.</summary>
     internal static partial class PathInternal
     {
-        private static readonly bool s_isCaseSensitive = GetIsCaseSensitive();
-
         /// <summary>Returns a comparison that can be used to compare file and directory names for equality.</summary>
         internal static StringComparison StringComparison
         {
             get
             {
-                return s_isCaseSensitive ?
+                return IsCaseSensitive ?
                     StringComparison.Ordinal :
                     StringComparison.OrdinalIgnoreCase;
             }
         }
 
         /// <summary>Gets whether the system is case-sensitive.</summary>
-        internal static bool IsCaseSensitive { get { return s_isCaseSensitive; } }
-
-        /// <summary>
-        /// Determines whether the file system is case sensitive.
-        /// </summary>
-        /// <remarks>
-        /// Ideally we'd use something like pathconf with _PC_CASE_SENSITIVE, but that is non-portable,
-        /// not supported on Windows or Linux, etc. For now, this function creates a tmp file with capital letters
-        /// and then tests for its existence with lower-case letters.  This could return invalid results in corner
-        /// cases where, for example, different file systems are mounted with differing sensitivities.
-        /// </remarks>
-        private static bool GetIsCaseSensitive()
+        internal static bool IsCaseSensitive
         {
-            try
-            {
-                string pathWithUpperCase = Path.Combine(Path.GetTempPath(), "CASESENSITIVETEST" + Guid.NewGuid().ToString("N"));
-                using (new FileStream(pathWithUpperCase, FileMode.CreateNew, FileAccess.ReadWrite, FileShare.None, 0x1000, FileOptions.DeleteOnClose))
-                {
-                    string lowerCased = pathWithUpperCase.ToLowerInvariant();
-                    return !File.Exists(lowerCased);
-                }
-            }
-            catch
+            get
             {
-                // In case something goes wrong (e.g. temp pointing to a privilieged directory), we don't
-                // want to fail just because of a casing test, so we assume case-insensitive-but-preserving.
-                return false;
+#if MS_IO_REDIST
+                return false; // Windows is always case-insensitive
```

Comment:
Is this going to break if I pass `FILE_FLAG_POSIX_SEMANTICS` to `CreateFile` or if I modify a folder to set it as case insensitive using `fsutil` (which is supported since Win10 17134)?

There are multiple ways to force windows to be case sensitive, including for the WSL mounted drive, which is fully browsable via File explorer and can be interacted with via the file system APIs

Reply:
https://devblogs.microsoft.com/commandline/per-directory-case-sensitivity-and-wsl/ goes into some more details about various changes that went in, but even prior to this NTFS has always supported case sensitivity. Its just that Windows hadn't handled it as "first class" if you enabled it.

Reply:
It will be just as broken as before (when we probed the Temp folder, which is not a good proxy for the behavior of a _specific_ directory - neither on Unix nor on Windows), but faster.

Reply:
This has been a global setting, independent on the path. This logic is broken if you have parts of the file system configured differently. This PR does not change it. Yes, it would be nice to fix that.

The unmanaged side of the runtime has similar global setting. It is larger problem than just what you can see in the managed libraries.